src/MicrosoftAi/AbstractionMapper.cs

```diff
@@ -0,0 +1,321 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Microsoft.Extensions.AI;
+using OllamaSharp.Models.Chat;
+
+namespace OllamaSharp.MicrosoftAi;
+
+/// <summary>
+/// Provides mapping functionality between OllamaSharp and Microsoft.Extensions.AI models.
+/// </summary>
+public static class AbstractionMapper
+{
+	/// <summary>
+	/// Maps a <see cref="ChatRequest"/> and <see cref="ChatDoneResponseStream"/> to a <see cref="ChatCompletion"/>.
+	/// </summary>
+	/// <param name="request">The chat request containing initial request data.</param>
+	/// <param name="response">The response stream with completion data.</param>
+	public static ChatCompletion? ToChatCompletion(ChatRequest request, ChatDoneResponseStream? response)
+	{
+		if (response is null)
+			return null;
+
+		var chatMessage = ToChatMessage(response.Message);
+		var completion = new ChatCompletion(chatMessage)
+		{
+			FinishReason = ToFinishReason(response.DoneReason),
+			AdditionalProperties = ParseOllamaChatResponseProps(response),
+			Choices = [chatMessage],
+			CompletionId = response.CreatedAt,
+			CreatedAt = DateTimeOffset.TryParse(response.CreatedAt, CultureInfo.InvariantCulture, DateTimeStyles.None, out var createdAt) ? createdAt : null,
+			ModelId = response.Model ?? request.Model,
+			RawRepresentation = response,
+			Usage = ParseOllamaChatResponseUsage(response)
+		};
+
+		return completion;
+	}
+
+	/// <summary>
+	/// Converts Microsoft.Extensions.AI messages and options to an OllamaSharp chat request.
+	/// </summary>
+	/// <param name="apiClient">The API client used for communication.</param>
+	/// <param name="chatMessages">A list of chat messages.</param>
+	/// <param name="options">Optional chat options to configure the request.</param>
+	/// <param name="stream">Indicates if the request should be streamed.</param>
+	public static ChatRequest ToOllamaSharpChatRequest(IOllamaApiClient apiClient, IList<ChatMessage> chatMessages, ChatOptions? options, bool stream)
+	{
+		// unused ChatOptions properties
+		// options.MaxOutputTokens,
+		// options.ToolMode,
+
+		return new ChatRequest
+		{
+			Format = options?.ResponseFormat == ChatResponseFormat.Json ? "json" : null,
```

Comment:
I think this should be:

```suggestion
			Format = options?.ResponseFormat is ChatResponseFormat.Json ? "json" : null,
```

... because then it will still work even if the app is trying to specify a JSON schema. I know Ollama won't respect the JSON schema but at least it will understand the app is asking for JSON.

Reply:
Hm, it says "CS9135: A constant value of type 'ChatResponseFormat' is expected"